### PR TITLE
feat: add display on org page field in AdditionalMetadata

### PR DESCRIPTION
**Pull request by DawoudSheraz**
### Description
Add display on org page checkmark under AdditionalMetadata fields (only displayed for ExecEd types). Since the field is boolean, there is no need to update the config `ADDITIONAL_METADATA_REQUIRED_FIELDS`. 

Related to PR https://github.com/openedx/course-discovery/pull/4021

### Testing
- Pull latest discovery and run migrations. Ensure you have one ExecEd 2u type course
- Checkout this branch for publisher
- Run publisher and visit the execEd course
- You should see the checkmark right under Additional Metadata section
- Toggle the value and ensure the backend is updated as expected

### Screenshots

Screenshot 2023-07-25 at 12 54 12 PM

Screenshot 2023-07-25 at 12 59 32 PM

### CODE REVIEW

File: src/components/EditCoursePage/__snapshots__/EditCoursePage.test.jsx.snap [hunk 1]
```diff
@@ -574,6 +574,7 @@ exports[`EditCoursePage renders page correctly with courseInfo 1`] = `
             "certificate_info_blurb": "blurb",
             "certificate_info_heading": "heading",
             "course_term_override": undefined,
+            "display_on_org_page": undefined,
```
Comment (AfaqShuaib09) [inline src/components/EditCoursePage/__snapshots__/EditCoursePage.test.jsx.snap:577]: don't its value need to be `true/false` as it's a boolean attribute?

File: src/components/EditCoursePage/AdditionalMetadataFields.jsx [hunk 2]
```diff
@@ -41,6 +41,19 @@ const AdditionalMetadataFields = (props) => {
...
+            type="checkbox"
+            label="additional_metadata.display_on_org_page"
+            disabled={disabled}
+            component="input"
+            required={isRequiredField('display_on_org_page')}
+            className="float-left mt-1"
+          />
+          <FieldLabel text="Display on Organization Page" className="font-weight-normal ml-2 float-left" />
+          <br />
+        </div>
```
Comment (AfaqShuaib09) [inline src/components/EditCoursePage/AdditionalMetadataFields.jsx:56]: I believe margin or some spacing would be good here

Screenshot 2023-07-24 at 5 31 22 PM

Comment (DawoudSheraz) [reply src/components/EditCoursePage/AdditionalMetadataFields.jsx:56]: Sure.

**Review by AliAdnanSohail — APPROVED** [review]
just a small suggestion

File: src/components/EditCoursePage/index.jsx [hunk 3]
```diff
@@ -277,6 +277,8 @@ class EditCoursePage extends React.Component {
       ],
       start_date: courseData.additional_metadata.start_date,
       end_date: courseData.additional_metadata.end_date,
+      // eslint-disable-next-line max-len
+      display_on_org_page: courseData.additional_metadata.display_on_org_page !== undefined ? courseData.additional_metadata.display_on_org_page : true,
```
Comment (AliAdnanSohail) [inline src/components/EditCoursePage/index.jsx:281]: nit:
```suggestion
      display_on_org_page: courseData.additional_metadata.display_on_org_page || true,
```

**Review by AliAdnanSohail — APPROVED** [review]

File: src/components/EditCoursePage/index.jsx (hunk 3 above)
Comment (DawoudSheraz) [reply src/components/EditCoursePage/index.jsx:281]: I did this first. But in the case of an unchecked box (False), this condition evaluates to True.

Comment (AliAdnanSohail) [reply src/components/EditCoursePage/index.jsx:281]: Ah! good point

**Review by AfaqShuaib09 — APPROVED** [review]

File: src/components/EditCoursePage/index.jsx (hunk 3 above)
Comment (DawoudSheraz) [reply src/components/EditCoursePage/index.jsx:281]: Ah, I was not aware of this operator.

**Pull request merged**